### PR TITLE
Raise docker-py requirement to 4.x

### PR DESCRIPTION
**Pull request by teodormadan**
Signed-off-by: Teo Madan <teodor.madan@nxp.com>

### CODE REVIEW

**Comment by teodormadan** [comment]
Compatibility has been addressed with https://github.com/ros-tooling/cross_compile/pull/237/, 

Still installing ros_cross_compile from source would bring older docker-py

**Comment by teodormadan** [comment]
Handled also the new BuildError 3.x docker API according to detected docker version, and remove the old test for BuildError api as not applicable.

File: setup.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
         package_name: ['docker/*.*', 'mixins/*.*'],
     },
     install_requires=[
-        'docker>=2,<3',
+        'docker>=2,<5',
```
Comment (emersonknapp) [inline setup.py:54]: I've been thinking about upgrading this, the old version requirement is from an old situation that no longer applies. Perhaps we could just make this

```suggestion
        'docker==4.*',
```

**Review by MichaelOrlov — APPROVED** [review]

**Review by MichaelOrlov — CHANGES_REQUESTED** [review]
@teodormadan @emersonknapp CI jobs for MacOS fails with this PR.
Please help to fix.

**Pull request closed without merging**